lib/jirawatch/cli/track.rb

```diff
@@ -6,55 +6,84 @@ class Track < Dry::CLI::Command
+
+        reader = TTY::Reader.new
+        reader.on(:keyctrl_p) do
+          if time_paused_at == 0
+            time_paused_at = Time.now
+            partial_time_spent = partial_time_spent + (time_paused_at - time_restarted_at)
+
+            time_unit =
+                if (partial_time_spent / 60).floor == 1
+                  "minute"
+                else
+                  "minutes"
+                end
 
-        begin
-          loop do
-            sleep 1
+            puts
+            puts "Timer has been paused with #{(partial_time_spent / 60).floor} #{time_unit} logged"
+          else
+            time_paused_at = 0
+            time_restarted_at = Time.now
+            puts
+            puts "Timer has been restarted"
           end
+        end
+
+        loop do
+          reader.read_line("")
         rescue Interrupt
-          time_spent = (Time.now - started_at).floor
+          total_time_spent = (partial_time_spent + (Time.now - time_restarted_at)).floor
 
           # Jira work logs have minute sensitivity thus API calls will fail with a time spent
           # which is less than 60 seconds
 
-          fail! "Jira can't log less than 60 seconds of work" if time_spent < 60
+          fail! "Jira can't log less than 60 seconds of work" if total_time_spent < 60
 
-          worklog_file = Jirawatch.configuration.template_worklog_file % started_at.to_i
+          worklog_file = Jirawatch.configuration.template_worklog_file % time_started_at.to_i
           worklog_lines = []
 
           File.write worklog_file, options.fetch(:worklog_message) if options.key? :worklog_message
           TTY::Editor.open(
               worklog_file,
               content: "\n" +
-                  "# You spent #{(time_spent / 60).floor} minutes on this issue\n" +
+                  "# You spent #{(total_time_spent / 60).floor} minutes on this issue\n" +
                   "# Write here your work log description\n" +
                   "# If you leave this empty, no work log will be saved"
           ) unless options.key? :worklog_message
 
-          File.readlines(Jirawatch.configuration.template_worklog_file % started_at.to_i).each do |line|
+          File.readlines(Jirawatch.configuration.template_worklog_file % time_started_at.to_i).each do |line|
             worklog_lines << line unless line.start_with?("#") or line.strip.empty?
           end
 
           @jira_client.Issue.find(issue_key).worklogs.build.save(
               {
-                  timeSpentSeconds: time_spent,
-                  started: started_at.strftime("%Y-%m-%dT%H:%M:%S.%L%z"),
+                  timeSpentSeconds: total_time_spent,
+                  started: time_started_at.strftime("%Y-%m-%dT%H:%M:%S.%L%z"),
                   comment: worklog_lines.join("\n")
               }
           ) unless worklog_lines.empty?
 
           puts "Worklog was empty, time was not tracked" if worklog_lines.empty?
 
           File.delete worklog_file
+          exit
```

Comment:
You don't need `exit` here

Reply:
You are right! There was an error in the loop statement